libraries/AP_Baro/examples/ICM20789/ICM20789.cpp

```diff
@@ -0,0 +1,291 @@
+/*
+  minimal test program for ICM20789 baro with IMU on SPI and baro on I2C
+ */
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_HAL/I2CDevice.h>
+#include <AP_Baro/AP_Baro.h>
+#include <AP_BoardConfig/AP_BoardConfig.h>
+#include <utility>
+#include <stdio.h>
+
+const AP_HAL::HAL& hal = AP_HAL::get_HAL();
+
+static AP_HAL::OwnPtr<AP_HAL::Device> i2c_dev;
+static AP_HAL::OwnPtr<AP_HAL::Device> spi_dev;
+static AP_HAL::OwnPtr<AP_HAL::Device> lidar_dev;
+
+// SPI registers
+#define MPUREG_WHOAMI                           0x75
+#define MPUREG_USER_CTRL                        0x6A
+#define MPUREG_PWR_MGMT_1                       0x6B
+
+#define MPUREG_INT_PIN_CFG                      0x37
+#       define BIT_BYPASS_EN                        0x02
+#       define BIT_INT_RD_CLEAR                     0x10    // clear the interrupt when any read occurs
+#       define BIT_LATCH_INT_EN                     0x20    // latch data ready pin
+
+#define BIT_USER_CTRL_I2C_MST_EN             0x20 // Enable MPU to act as the I2C Master to external slave sensors
+#define BIT_PWR_MGMT_1_DEVICE_RESET          0x80 // reset entire device
+#define BIT_USER_CTRL_I2C_IF_DIS             0x10 // Disable primary I2C interface and enable hal.spi->interface
+#define BIT_PWR_MGMT_1_CLK_XGYRO             0x01 // PLL with X axis gyroscope reference
+#define BIT_PWR_MGMT_1_CLK_ZGYRO             0x03 // PLL with Z axis gyroscope reference
```

Comment:
So many beautiful macros you don't use anywhere :stuck_out_tongue:

Reply:
They're cheap - and might be useful to someone in the future...